src/util/documentUtils.ts

```diff
@@ -0,0 +1,86 @@
+import * as vscode from 'vscode';
+import type { CodeBlock, SerializedRange } from '../types';
+
+const CODE_BLOCK_STARTS = new Set(['```python', '```py', '```groovy']);
+const CODE_BLOCK_END = '```';
+
+/**
+ * Parse code blocks from a Markdown document that can be run by Deephaven.
+ * @param document The document to parse code blocks from.
+ * @returns Code blocks that can be run by Deephaven.
+ */
+export function parseMarkdownCodeblocks(
+  document: vscode.TextDocument
+): CodeBlock[] {
+  const lines = document.getText().split('\n');
+
+  const codeBlocks: CodeBlock[] = [];
+  let startPos: vscode.Position | null = null;
+  let languageId = '';
+
+  // Create ranges for each code block in the document
+  for (let i = 0; i < lines.length; ++i) {
+    const line = lines[i];
+
+    // Start of Deephaven supported code block
+    if (CODE_BLOCK_STARTS.has(line)) {
```

Comment:
Not directly related to your change, but in a lot of our docs, we have a code block and have more options after the language, e.g.
````
```groovy order=crossJoin,leftOuterJoin,fullOuterJoin
````
Right now we're not matching those blocks. I guess to properly get them you could use some sort of regex of a language followed by whitespace or comma or something.